node/src/Pos/Wallet/Web/State/Storage.hs

```diff
@@ -269,17 +273,21 @@ getWalletAddresses =
     view wsWalletInfos
 
 getAccountWAddresses :: AddressLookupMode
+                  -> NeedSorting
                   -> AccountId
                   -> Query (Maybe [CWAddressMeta])
-getAccountWAddresses mode accId =
+getAccountWAddresses mode (NeedSorting needSorting) accId =
     withAccLookupMode mode (fetch aiAddresses) (fetch aiRemovedAddresses)
   where
+    sorting
+        | needSorting = sortOn adiSortingKey
+        | otherwise = identity
     fetch :: MonadReader WalletStorage m => Lens' AccountInfo CAddresses -> m (Maybe [CWAddressMeta])
     fetch which = do
         cAddresses <- preview (wsAccountInfos . ix accId . which)
         -- here `cAddresses` has type `Maybe CAddresses`
         pure $
-            (map adiCWAddressMeta . sortOn adiSortingKey . map snd . HM.toList)
+            (map adiCWAddressMeta . sorting . toList)
```

Comment:
So we would not `map adiCWAddressMeta`, and that'd keep the `adiSortingKey` info around, so the caller can sort the result if it really needs to.